include/boost/geometry/algorithms/detail/overlay/enrich_intersection_points.hpp

```diff
@@ -396,16 +397,36 @@ inline void enrich_intersection_points(Turns& turns,
             std::vector<indexed_turn_operation>
         >;
 
-    // From here on, turn indexes are used (in clusters, next_index, etc)
-    // and turns may not be DELETED - they may only be flagged as discarded
-    discard_duplicate_start_turns(turns, geometry1, geometry2);
+    // Turns are often used by index (in clusters, next_index, etc)
+    // and turns may therefore NOT be DELETED - they may only be flagged as discarded
 
     bool has_cc = false;
-    bool const has_colocations
-        = detail::overlay::handle_colocations
+    bool has_colocations = false;
+
+    if (BOOST_GEOMETRY_CONDITION (! is_buffer))
+    {
+        // Handle colocations, gathering clusters and (below) their properties.
+        has_colocations = detail::overlay::handle_colocations
+                    <
+                        Reverse1, Reverse2, OverlayType, Geometry1, Geometry2
+                    >(turns, clusters, robust_policy);
+        // Gather cluster properties (using even clusters with
+        // discarded turns - for open turns)
+        detail::overlay::gather_cluster_properties
             <
-                Reverse1, Reverse2, OverlayType, Geometry1, Geometry2
-            >(turns, clusters, robust_policy);
+                Reverse1,
+                Reverse2,
+                OverlayType
+            >(clusters, turns, target_operation,
```

Comment:
`target_operation` is known at compile-time. It could be passed as template parameter and then checked with `if BOOST_GEOMETRY_CONSTEXPR` inside `gather_cluster_properties`. But a different PR would probably be better for this.

Reply:
Will do it in another PR indeed. Not today.